app/src/backend/environmentVariables/index.ts

```diff
@@ -0,0 +1,40 @@
+import firebaseApp from "firebase";
+import { deleteField, doc, getFirestore, setDoc, updateDoc } from "firebase/firestore";
+
+const getDocPath = (ownerId: string) => {
+  const db = getFirestore(firebaseApp);
+  return doc(db, "environmentVariables", ownerId);
+};
+
+export const setEnvironmentVariable = async (
+  ownerId: string,
+  payload: {
+    newVariable: {
+      key: string;
+      value: string | number | boolean;
+    };
```

Comment:
Use the same type `Record<string, EnvironmentVariableValue>` and make `localValue` optional so that it isn't synced to firebase